src/components/Dashboard.tsx

```diff
@@ -94,38 +85,81 @@ const TemplateComponent: React.FC<{ template: TemplateInList }> = ({ template })
   )
 }
 
-const FilterComponent: React.FC<{
+const PanelComponent: React.FC<{
   template: TemplateDetails
-  filter: Filter
-  setFiltersCount: Function
-}> = ({ template, filter, setFiltersCount: setLoadingFilters }) => {
+  filters: Filter[]
+}> = ({ template, filters }) => {
+  const { preferences } = usePrefs()
   const templateType = template.code
-  const { loading, applicationCount } = useListApplications({
+  const [loadedFiltersCount, setLoadedFiltersCount] = useState(0)
+  const [totalMatchFilter, setTotalMatchFilter] = useState<{ [key: number]: number }>(
+    filters.reduce((totalPerFilter, filter) => ({ ...totalPerFilter, [filter.id]: 0 }), {})
+  )
+
+  // Convert to string - Default is 100 is nothing is preset on preferences
+  const perPage = preferences?.dashboardFetchedApplications || '100'
```

Comment:
I'm not sure why we'd want to paginate this. Isn't the dashboard query to determine the total number of each type (filter type) of application? So what would be the purpose of not fetching the full set?

Reply:
Just that the `getApplicationsList` uses the pagination. I don't think it's simple to remove/separate a new query to not use it

Reply:
It was already using it. I didn't change, and tried to keep with the minimal changes possible! Although I agree that it would be nicer if we could do for all applications - Just one things that I might consider - we also do filters for Approved/Rejected. So it would basically query for all existing in all history of the App

Reply:
Ah okay, in which case we should just set a default page size in `getApplicationList` that is really big (something we'll never reach). It shouldn't be a preference, cos we basically want to "fetch all". 

Actually, it's probably not that hard to make the "numberToFetch" parameter optional and only specify a `first` value if defined. I dunno the easiest way, but basically the dashboard query needs to get ALL matching applications somehow.

Reply:
I might be misunderstanding slightly, but one problem with this approach is that we're filtering on the front-end, which means the returned list of applications could be enormous.

If we have 10,000 applications in the database, one of these filters (e.g. "can start review") is likely to reduce that list down to a handful (the current reviewers' available work). But the way it is here, you'd be pulling down all 10,000 and then filtering on the front-end.

Is that correct, or am I following it wrong?

I was thinking that we'd still filter on the back-end, but we'd combine the filters into a single gql filter object. So we'd still be retrieving a "reasonable" size list of applications, then just *sorting* them by filter-type in the front-end. 

Does that make sense?

Reply:
Thanks. I think that's much better. : )

And I noticed that after Andrei's change of indexes the App is faster, but the Dashboard should also benefit from this change. Also good call you made purposing this solution.